### PR TITLE
New serverless pattern - appsync-private-api-sam

### PR DESCRIPTION
**Pull request by Tessot**
Issue #, if available:

Description of changes: This pattern shows how you can deploy an AWS AppSync Private API which can only be invoked by resources within your private network.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: appsync-private-api-sam/example-pattern.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
+{
+  "title": "AWS AppSync Private API ",
+  "description": "Create an AWS AppSync Private API with a sample API to demonstrate how you can invoke Private API from resources in your private network",
+  "language": "Yaml",
```
Comment (pputhran) [inline appsync-private-api-sam/example-pattern.json:4]: This should be YAML (uppercase)

**Review by pputhran — CHANGES_REQUESTED** [review]

File: appsync-private-api-sam/README.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+5. Note the outputs from the SAM deployment process. Two of the outputs `AppSync GraphQL API URL` and `AppSync VPC Endpoint DNS` will be used to test this pattern.
+
+## How it works
+
+This patterns creates and AppSync Interface VPC Endpoint and a sample AppSync Private API backed with a DynamoDB data source. Requests to AppSync Private APIs will go through AWS’s private network without going over the internet. GraphQL requests from your application are routed via the interface VPC endpoint to AppSync Private API. Interface VPC endpoint is powered by [AWS PrivateLink](https://aws.amazon.com/privatelink/), a highly available, scalable technology that enables you to privately connect your VPC to AWS services like AWS AppSync as if the services were in your VPC.
+
+API Key is used as the authorization mode for the AppSync API however it is not recommended to use API Key for production application, kindly refer to other authorization modes supported by AppSync in the [documentation](https://docs.aws.amazon.com/appsync/latest/devguide/security-authz.html)
```
Comment (bfreiberg) [inline appsync-private-api-sam/README.md:53]:
```suggestion
API Key is used as the authorization mode for the AppSync API. However it is not recommended to use API Key for production application, please refer to other authorization modes supported by AppSync in the [documentation](https://docs.aws.amazon.com/appsync/latest/devguide/security-authz.html)
```

File: appsync-private-api-sam/README.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+## Testing
+
+You can easily test this pattern using any command prompt that supports the `curl` command. Refer to the outputs `AppSyncApiUrl`, `AppSyncApiKey` and `AppSyncVPCEndpointDNS` from deploying the SAM application which will be used for testing.
```
Comment (bfreiberg) [inline appsync-private-api-sam/README.md:57]:
```suggestion
You can test this pattern using any command prompt that supports the `curl` command. Refer to the outputs `AppSyncApiUrl`, `AppSyncApiKey` and `AppSyncVPCEndpointDNS` from deploying the SAM application which will be used for testing.
```

File: appsync-private-api-sam/README.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+-- Using AppSync GraphQL API URL (enabled by Private DNS settings = Yes)
+
+```curl {AppSyncGraphQLAPIURL} \
+   -H "Content-Type:application/graphql" \
+   -H "x-api-key:da2-{AppSyncApiKey}" \
+   -d '{"query": "query MyQuery {listRestaurants {items {name state restaurantId zip cuisine }}}","variables":"{}"}'
+```
+
+-- Using AppSync VPC Interface Endpoint DNS (you will need to pass the`AppSyncGraphQLAPIURL` in the host header, remember to remove suffix `www.`)
```
Comment (bfreiberg) [inline appsync-private-api-sam/README.md:73]: Do you mean prefix instead?

**Review by bfreiberg — CHANGES_REQUESTED** [review]

**Comment by bfreiberg** [comment]
Hi, do you still plan to address the requested changes?

**Comment by Tessot** [comment]
I will be creating a new pull request with the changes

**Pull request closed without merging**